records_mover/db/loader.py

```diff
@@ -0,0 +1,105 @@
+from ..records.records_format import BaseRecordsFormat, DelimitedRecordsFormat
+from ..records.types import RecordsFormatType
+from ..records.load_plan import RecordsLoadPlan
+from ..records.records_directory import RecordsDirectory
+from ..url.filesystem import FilesystemDirectoryUrl
+from contextlib import contextmanager
+from ..url.base import BaseDirectoryUrl
+from tempfile import TemporaryDirectory
+from abc import ABCMeta, abstractmethod
+from typing import Optional, Type, Iterator, IO, List
+import sqlalchemy
+
+
+class NegotiatesLoadFormat(metaclass=ABCMeta):
+    @abstractmethod
+    def best_records_format(self) -> BaseRecordsFormat:
+        ...
+
+    # TODO: Should these be part of other interface?  nothing to do with negotiation
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker